### PR TITLE
Refactoring global logger

### PR DESCRIPTION
**Pull request by ajyl**
This PR is dependent on #15 , will need to be refactored once #15 is merged.
Removes all calls to `global logger`.

### CODE REVIEW

**Review by davidjurgens — CHANGES_REQUESTED** [review]

File: potato/flask_server.py [hunk 1]
```diff
@@ -101,6 +101,18 @@
...
+config.update({
+    "verbose": args.verbose,
+    "very_verbose": args.very_verbose,
+    "__debug__": args.debug,
+    "__config_file__": args.config_file,
+})
+init_logger(config)
+logger = logging.getLogger(config.get("logger_name", "potato"))
+breakpoint()
+
```
[the remark this answers is not in the document]
Comment (ajyl) [reply potato/flask_server.py:115]: Hmm, I thought I had removed this -- removing now...

File: potato/server_utils/logger.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import logging
+
+
+def init_logger(config):
+    """
+    Initialize logger.
+    """
+    logger_name = config.get("logger_name", "potato")
+    logger = logging.getLogger(logger_name)
```
[the remark this answers is not in the document]
Comment (ajyl) [reply potato/server_utils/logger.py:13]: Hmm, the nice thing about `config` was that we're calling `config.update()`, such that when `init_config()` is being called, the global variable was being updated.

With `logger`, the issue I'm running into is as following:

1) `logger` is initialized as a global variable with something like `logger = None`
2) In `flask_server.py`, we do `import logger`. Note that at this point, `logger` is None.
3) Eventually, `init_logger()` is called, but unlike our `config`, which is updating our `config` variable, this line does not update `logger` for us.
4) This is a problem `logger` in `flask_server.py` stays as `None` despite calling `init_logger()`.

I agree that a solution like the one we had for `config` would be nicer. Furthermore, previously, my understanding is that a single logger instance is being created by `config["logger_name"]` -- having a dependency on this config value makes it a bit trickier to refactor. Any thoughts on removing such dependency and just using module names? https://stackoverflow.com/questions/15727420/using-logging-in-multiple-modules

[the remark this answers is not in the document]
Comment (davidjurgens) [reply potato/server_utils/logger.py:13]: Hmm... I see the issue now. How about we merge this one so it Just Works™ for now and decide on logging more later. I had thought about doing module name-based stuff too (coming from Java land where that's standard) but didn't bother with it here. Now that there's actually submodules, I think it's reasonable if we wanted to use it.

File: potato/flask_server.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajyl) [reply potato/flask_server.py:115]: If we remove the dependency between `logger` and `config`, we can certainly move everything related to `args` to `run_server()` and simply initialize `logger` as `getLogger(__name__)` at the module level, which I think might be the cleanest approach?

File: potato/server_utils/logger.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ajyl) [reply potato/server_utils/logger.py:13]: My vote would be to break the dependency between `logger` and `config` and change all `loggers` to module-level loggers in this PR, breaking that dependency will likely make all future refactors easier in my opinion. See my [response](https://github.com/davidjurgens/potato/pull/16#discussion_r1028620862) to your other comment!

File: potato/flask_server.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidjurgens) [reply potato/flask_server.py:114]: sounds great! (see other comment too)

File: potato/server_utils/logger.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidjurgens) [reply potato/server_utils/logger.py:13]: Agreed. let's break the dependency and switch to module-level logging. That will probably work in our favor for end-users easy filtering out some detailed messages at the module level

**Comment by ajyl** [comment]
Decoupled the logger from config per our discussion! I believe this is ready to be merged.

**Review by davidjurgens — APPROVED** [review]

**Pull request merged**